src/ArmTemplates/Extractor/Utilities/DataProcessors/GroupDataProcessor.cs

```diff
@@ -0,0 +1,64 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Groups;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors.Absctraction;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors
+{
+    public class GroupDataProcessor: IGroupDataProcessor
+    {
+        public IDictionary<string, string> OverrideRules {
+            get => this.overrideRules;  
+            set => this.overrideRules = value;
+        }
+
+        IDictionary<string, string> overrideRules;
+
+        public GroupDataProcessor() {
+            this.OverrideRules = new Dictionary<string, string>() {
+                { "Administrators", "administrators" },
+                { "Developers", "developers" },
+                { "Guests", "guests" }
+            };
+        }
+
+        public void ProcessData(List<GroupTemplateResource> groupTemplates, ExtractorParameters extractorParameters) {
+            
+            foreach(var groupTemplate in groupTemplates) 
+            {
+                if (extractorParameters.OverrideGroupNames)
+                {
+                    this.OverrideName(groupTemplate);
+                }
+            }
+            
+        }
+
+        public void OverrideName(GroupTemplateResource template)
+        {
+            if (this.OverrideRules == null)
+            {
+                return;
+            }
+
+            foreach (var rule in this.OverrideRules)
+            {
+                if (template.Properties.DisplayName.Equals(rule.Key))
```

Comment:
Imagine such a situation. In `group` entity I have `name` and `description`.
What if we need a rule for overriding `Administrators` name to smth, and a rule for overriding `Administrators` description to smth. Then you get conflicting key in a dictionary, which is a runtime-error (adding same keys to dictionary).

I would recommend wrapping the rule into a separate class, where you define both `PropertyType` or smth like that (I don't like this naming, but it is honest work :D) and a value you need to override. So it would be:
```
class OverrideRule
{
    public PropertyType PropertyType {get;set;}
    public string value {get;set;}
    
    public override bool Equals() { ... }
    public string GetHashCode() { ... }
}
```
and you have also to define `PropertyType` enum:
```
enum PropertyType
{
    Name,
    Description
}
```
That is a lot of classes and extra work, but we are having a full control over the override rules. 

What do you think?

Reply:
Lets do it this way for a moment and if there is a need to make it more generic, then we will proceed to that